src/Core/Email/EmailDataConfigurator.php

```diff
@@ -125,4 +118,23 @@ public function validateConfiguration(array $config)
             $config['smtp_config']['password']
         );
     }
+    */
+
+    /**
+     * @return OptionsResolver
+     */
+    protected function buildResolver(): OptionsResolver
+    {
+        $resolver = (new OptionsResolver())
+            ->setDefined(self::CONFIGURATION_FIELDS)
+            ->setAllowedTypes('send_emails_to', 'int')
+            ->setAllowedTypes('mail_method', 'int')
+            ->setAllowedTypes('mail_type', 'int')
+            ->setAllowedTypes('log_emails', 'bool')
+            ->setAllowedTypes('dkim_enable', 'bool')
+            ->setAllowedTypes('smtp_config', 'array')
```

Comment:
The `smtp_config` needs a sub resolver to validate its internal format
Here is an example of how you can do this:
https://github.com/PrestaShop/PrestaShop/blob/298b61ea121c7f6a7a35f7447e1fe91701a12904/src/PrestaShopBundle/Form/Admin/Type/EntitySearchInputType.php#L142-L144

https://github.com/PrestaShop/PrestaShop/blob/298b61ea121c7f6a7a35f7447e1fe91701a12904/src/PrestaShopBundle/Form/Admin/Type/EntitySearchInputType.php#L222-L244